plugins/action/k8s_info.py

```diff
@@ -92,22 +92,38 @@ def load_template(self, template, new_module_args, task_vars):
                     if value is not None and not isinstance(value, string_types):
                         raise AnsibleActionFail("%s is expected to be a string, but got %s instead" % (s_type, type(value)))
             try:
-                trim_blocks = boolean(template_args.get('trim_blocks', True), strict=False)
-                lstrip_blocks = boolean(template_args.get('lstrip_blocks', False), strict=False)
+                template_param.update(dict(trim_blocks=boolean(template_args.get('trim_blocks', True), strict=False)))
+                template_param.update(dict(lstrip_blocks=boolean(template_args.get('lstrip_blocks', False), strict=False)))
             except TypeError as e:
                 raise AnsibleActionFail(to_native(e))
 
-            newline_sequence = template_args.get('newline_sequence', self.DEFAULT_NEWLINE_SEQUENCE)
-            variable_start_string = template_args.get('variable_start_string', None)
-            variable_end_string = template_args.get('variable_end_string', None)
-            block_start_string = template_args.get('block_start_string', None)
-            block_end_string = template_args.get('block_end_string', None)
+            template_param.update(dict(newline_sequence=template_args.get('newline_sequence', self.DEFAULT_NEWLINE_SEQUENCE)))
+            template_param.update(dict(variable_start_string=template_args.get('variable_start_string', None)))
+            template_param.update(dict(variable_end_string=template_args.get('variable_end_string', None)))
+            template_param.update(dict(block_start_string=template_args.get('block_start_string', None)))
+            template_param.update(dict(block_end_string=template_args.get('block_end_string', None)))
+        else:
+            raise AnsibleActionFail("Error while reading template file - "
+                                    "a string or dict for template expected, but got %s instead" % type(template))
+        return template_param
+
+    def load_template(self, template, new_module_args, task_vars):
+        # template is only supported by k8s module.
+        if self._task.action not in ('k8s', 'kubernetes.core.k8s', 'community.okd.k8s'):
+            raise AnsibleActionFail("'template' is only supported parameter for 'k8s' module.")
+
+        template_params = []
+        if isinstance(template, string_types) or isinstance(template, dict):
+            template_params.append(self.get_template_args(template))
+        elif isinstance(template, list):
+            for element in template:
+                template_params.append(self.get_template_args(element))
         else:
             raise AnsibleActionFail("Error while reading template file - "
                                     "a string or dict for template expected, but got %s instead" % type(template))
 
         # Option `lstrip_blocks' was added in Jinja2 version 2.7.
-        if lstrip_blocks:
+        if any([tmp['lstrip_blocks'] for tmp in template_params]):
```

Comment:
- I think you can move this block in a new function just to isolate the logic.
- Core has a similar block that also handle Jinja2 <2.7: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/template.py#L66-L76